pkg/receive/handler.go

```diff
@@ -132,43 +144,6 @@ func (h *Handler) testReadyHandler(f http.Handler) http.HandlerFunc {
 	return h.testReady(f.ServeHTTP)
 }
 
-// Run serves the HTTP endpoints.
-func (h *Handler) Run(ctx context.Context) error {
-	level.Info(h.logger).Log("msg", "Start listening for connections", "address", h.options.ListenAddress)
-
-	listener, err := net.Listen("tcp", h.options.ListenAddress)
-	if err != nil {
-		return err
-	}
-
-	// Monitor incoming connections with conntrack.
-	listener = conntrack.NewListener(listener,
-		conntrack.TrackWithName("http"),
-		conntrack.TrackWithTracing())
-
-	operationName := nethttp.OperationNameFunc(func(r *http.Request) string {
-		return fmt.Sprintf("%s %s", r.Method, r.URL.Path)
-	})
-	mux := http.NewServeMux()
-	mux.Handle("/", h.router)
-
-	errlog := stdlog.New(log.NewStdlibAdapter(level.Error(h.logger)), "", 0)
-
-	httpSrv := &http.Server{
-		Handler:  nethttp.Middleware(opentracing.GlobalTracer(), mux, operationName),
-		ErrorLog: errlog,
-	}
-
-	var g run.Group
```

Comment:
@brancz IIUC it was running in separate group than the rest of the actors? that probably caused the hanging in SIGINT

Reply:
Yes you're probably right.